plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/CachingJdbcClient.java

```diff
@@ -111,25 +140,22 @@ public CachingJdbcClient(
         this.cacheMissing = cacheMissing;
         this.identityMapping = requireNonNull(identityMapping, "identityMapping is null");
 
-        EvictableCacheBuilder<Object, Object> cacheBuilder = EvictableCacheBuilder.newBuilder()
-                .expireAfterWrite(metadataCachingTtl.toMillis(), MILLISECONDS)
+        long cacheSize = metadataCachingTtl.equals(CACHING_DISABLED)
+                // Disables the cache entirely
+                ? 0
+                : cacheMaximumSize;
+        Function<Duration, EvictableCacheBuilder<Object, Object>> cacheBuilder = cachingTtl -> EvictableCacheBuilder.newBuilder()
```

Comment:
Instead can we have a `Supplier` or just have builder and share them, since we are initializing a builder - maybe we could build schemaNamesCache and tableNamesCache separately once all the other caches are build - with custom ttl. WDYT ?

Reply:
Builder cannot be reused, those methods have to be called no more than once.

Reply:
Extracted to a separate method

Reply:
Supplier will mean we need to call the long `expireAfterWrite()` and repeat all the unit conversions multiple times.